R/layers.R

```diff
@@ -120,6 +120,31 @@ sass_layer <- function(
   )
 }
 
+
+#' @export
+#' @param file file path to a `.scss` file.
+#' @describeIn sass_layer Read in a `.scss` file with parse special `/*-- scss:(functions|defaults|rules|mixins) --*/` comments as relevant sections of a `sass_layer()`.
+sass_layer_file <- function(file) {
+  src <- readLines(file)
+  # https://github.com/quarto-dev/quarto-cli/blob/3d6063/src/command/render/sass.ts#L119-L123
+  pattern_key <- "scss:(functions|defaults|rules|mixins)"
+  pattern <- paste0("^/\\*--[ \\t]*", pattern_key, "[ \\t]*--\\*/$")
+  idx <- grep(pattern, src)
+  if (length(idx) == 0) {
+    stop("`file` must contain special comments of the form `/*-- scss:(functions|defaults|rules|mixins) --*/`")
```

Comment:
It just occurred to me that if we want to switch to using `sass_layer_file()` over `sass_file()` in `rmarkdown::html_document_base()`, then we should probably `return(sass_layer(rules = src))` (or similar) instead of erroring